### PR TITLE
PUD-1589: Add returned to custody letter to probation

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/NotInCustodyLetterToProbationGotenbergComponentTest.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        CroNumber("1234/56A"),
+        LocalDate.now()
+      )
+    )
+    updateRecallWithRequiredInformationForNotInCustodyLetterToProbation(
+      recall.recallId,
+      recallType = RecallType.FIXED
+    )
+
+    val letterToPrisonId = authenticatedClient.generateDocument(recall.recallId, LETTER_TO_PROBATION, FileName("BILLIE BOBS LETTER TO PROBATION.pdf"))
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/NotInCustodyLetterToProbationGotenbergComponentTest.kt:53]: `letterTo...` in `LetterToProbation` test

**Review by AndyNewton — COMMENTED** [review]
QQs ...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonConfirmationGenerator.kt [hunk 2]
```diff
@@ -12,11 +12,11 @@ class LetterToPrisonConfirmationGenerator(
 ) {
   fun generateHtml(context: LetterToPrisonContext): String =
     Context().apply {
-      setVariable("fullName", context.prisonerNameOnLicense)
+      setVariable("fullName", context.prisonerNameOnLicence)
 
       with(context.recall) {
         val recallDescription = RecallDescription(this.recallType(), this.recallLength)
-        setVariable("recallLengthDescription", recallDescription.asFixedTermLengthDescription())
+        setVariable("recallLengthDescription", recallDescription.asTitle())
```
[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/lettertoprison/LetterToPrisonConfirmationGenerator.kt:19]: in this case yes. Will be changed in forthcoming Letter to Prison ticket.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt [hunk 3]
```diff
@@ -31,13 +31,15 @@ import uk.gov.justice.digital.hmpps.managerecallsapi.controller.extractor.TokenE
...
 import uk.gov.justice.digital.hmpps.managerecallsapi.db.DocumentCategory.LETTER_TO_PRISON
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.DocumentCategory.LETTER_TO_PROBATION
 import uk.gov.justice.digital.hmpps.managerecallsapi.db.DocumentCategory.REASONS_FOR_RECALL
 import uk.gov.justice.digital.hmpps.managerecallsapi.db.DocumentCategory.RECALL_NOTIFICATION
 import uk.gov.justice.digital.hmpps.managerecallsapi.db.DocumentCategory.REVOCATION_ORDER
 import uk.gov.justice.digital.hmpps.managerecallsapi.documents.dossier.DossierService
 import uk.gov.justice.digital.hmpps.managerecallsapi.documents.dossier.ReasonsForRecallService
 import uk.gov.justice.digital.hmpps.managerecallsapi.documents.encodeToBase64String
 import uk.gov.justice.digital.hmpps.managerecallsapi.documents.lettertoprison.LetterToPrisonService
+import uk.gov.justice.digital.hmpps.managerecallsapi.documents.notincustodylettertoprobation.NotInCustodyLetterToProbationService
```
[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt:42]: The package is kind of relevant. The LetterToProbation* is part of recall notification,  the NotInCustodyLetterToProbation* is standalone. But happy to discuss class & package names.

**Review by AndyNewton — APPROVED** [review]

**Pull request merged**